jobboard/tests.py

```diff
@@ -182,3 +182,97 @@ def test_page_display_to_user_that_is_recruiter(client):
     jobs_to_display = Job.get_jobs_posted_on_or_after_specific_date(date_six_months_ago)
     # test that jobs displayed are only jobs created 6 months ago
     assert response.context['jobs_to_display'] == jobs_to_display
+
+
+@pytest.mark.django_db
+def test_job_display_to_user_that_is_student_not_applied(client, example_student_a):
+    client.force_login(example_student_a.user)
+    response = client.get("/job/1/")
+    assert response.status_code == 200
+    assertTemplateUsed(response, 'jobboard/job.html')
+    assert response.context['is_student'] is True
+    assert response.context['is_recruiter'] is False
+    # example_student_a haven't applied to job with id=1
+    assert response.context["is_student_applied"] is False
+    job_to_display = Job.objects.get(id=1)
+    assert response.context['job_data'] == job_to_display
+
+
+@pytest.mark.django_db
+def test_apply_button_for_student_trying_to_apply_and_test_after_apply(client, example_student_a):
```

Comment:
I think this test should probably be broken apart to multiple smaller tests. Usually you want to have only one HTTP requests per-test unless those requests are very tightly coupled.

Reply:
OK, no problem.  I'll split it

Reply:
@ifireball  I think here, you can call these requests tightly coupled.. I'm checking for the same user that he haven't applied for a job - checking the `is_student_applied = False`; Then, taking this same student to the page of applying to a job, checking if `is_student_applied = True` , and then, making this same student to manually trying to access the page of applying the same job again. 
 I don't want to save the application of that student to the DB, so I think its better all will be in one test.

Reply:
I think the code would still be easy to understand if those are separate tests. And it not necessary in practice that the HTTP calls you test will happen in the order you test them in.

Reply:
For HTTP calls to be coupled you need to have some information sharing mechanism to be passed between them like e.g. a cookie, you don't have that here.

Reply:
OK. I'll change it